### ISSUE TITLE
TestNearest does not work with Suite tests on Golang

### ISSUE THREAD

**Issue by lourenci** [vim-test/vim-test#624]
Using https://github.com/stretchr/testify, we're able to use a suite of tests. The lib helps us with test collisions, and it is being extensively used for the go community.

Using `vim-test` and `testify`, the `TestNearest` does not work with its purpose, and it runs the whole suite instead. Running a specific test using the suite differs from the conventional way: we've to pass the suite test together with the test to run (E.g.: `go test -run TestYourTestSuiteMethod ./package -testify.m TestYourIndividualMethod`).

Is there a way to extend the go strategy to deal with suite tests?

Thank you!

**Comment by codeinabox** [comment]
Should be possible, just a case of adding some new examples to `spec/gotest_spec.vim` and then going from there. Happy to give a hand with this

**Comment by yshadow** [comment]
Hello guys, this plugin is great I'm using it for table based tests, but as it already being said - it doesn't work for suite of tests. Calling :TestNearest gives you just:
testing: warning: no tests to run
PASS

Is there any progress with this one? Thank you

**Comment by yshadow** [comment]
Hi guys. Anything new regarding this? Thanks.

**Comment by FreedomKnight** [comment]
@msvitok77 I've implemented this feature. Can you help me check this PR https://github.com/vim-test/vim-test/pull/781 ? 
I needed use testify too, so I tried to support this.

**Comment by codeinabox** [comment]
The two testify PRs have been merged in, please let me know if this issue can now be closed

**Comment by msvitok77** [comment]
> @msvitok77 I've implemented this feature. Can you help me check this PR #781 ? I needed use testify too, so I tried to support this.

Hi @FreedomKnight. I can confirm it works with one small issue. When you define your suite with all the tests and then define a test like:
```
func Test_makeChunks(t *testing.T) {
    // your test code here
}
```
It assumes the suit to be that `Test_makeChunks` please take a look:
```
go test -v ./internal/service -run "Test_makeChunks$" -testify.m 
"Test_storage_Store"
=== RUN   Test_makeChunks
=== RUN   Test_makeChunks/valid_3_size_chunk_odd
=== RUN   Test_makeChunks/valid_3_size_chunk_even
=== RUN   Test_makeChunks/valid_3_size_chunk_less
--- PASS: Test_makeChunks (0.00s)
    --- PASS: Test_makeChunks/valid_3_size_chunk_odd (0.00s)
    --- PASS: Test_makeChunks/valid_3_size_chunk_even (0.00s)
    --- PASS: Test_makeChunks/valid_3_size_chunk_less (0.00s)
PASS
```
It's not a big deal as that test `Test_makeChunks` may be moved to another file, but just to let you know my testings ;-).

**Comment by FreedomKnight** [comment]
Do you mean this issue https://github.com/vim-test/vim-test/pull/782 ?
I think I've improved it after I metioned you.
It's already merged into master. Can you help to test in master branch?

**Comment by FreedomKnight** [comment]
Sorry, I forget to mention @msvitok77

**Comment by msvitok77** [comment]
> Do you mean this issue #782 ? I think I've improved it after I metioned you. It's already merged into master. Can you help to test in master branch?

Yes seems like that issue. And regarding the master branch - I've tested this on master branch. The latest commit I have is this: c83d5dffa054ac4663aea1eb7374f638b0462510

**Comment by FreedomKnight** [comment]
Yes, @msvitok77  this is the latest commit. My PR commit is the previous one.

**Comment by msvitok77** [comment]
> Yes, @msvitok77 this is the latest commit. My PR commit is the previous one.

Well in that case the issue #782 is still valid.

**Comment by FreedomKnight** [comment]
Sorry, as I'm not a native English speaker. Are you indicating that the issue is still present in your test case, or is it functioning correctly now?

**Comment by msvitok77** [comment]
> Sorry, as I'm not a native English speaker. Are you indicating that the issue is still present in your test case, or is it functioning correctly now?

So my testcase is like this:
1.) some_test.go 
     - contains only the testify suite with tests related to that suite - **works ok**
2.) some_test.go
     - contains testify suite with tests related to that suite + `some common test like func TestSomething(t *testing.T)` - **doesn't work**

**Comment by FreedomKnight** [comment]
In case2, which test case will be breaked in your example?

**Comment by FreedomKnight** [comment]
I use your example test
this is my result
<img width="816" alt="截圖 2024-01-09 晚上10 53 13" src="https://github.com/vim-test/vim-test/assets/1203922/13d64cfd-db10-4dd2-9b54-26765bd35a48">
<img width="1050" alt="截圖 2024-01-09 晚上10 53 46" src="https://github.com/vim-test/vim-test/assets/1203922/fe2c7d47-cc2b-4495-88af-09e6ce098f92">

**Comment by msvitok77** [comment]
> In case2, which test case will be breaked in your example?

As I've written above. The command constructed by this plugin looks like `go test -v ./internal/service -run "Test_makeChunks$" -testify.m 
"Test_storage_Store"`

And that `Test_makeChunks` is common test like func Test_makeChunks(t *testing.T) {....}`. While that `Test_storage_Store` is one of the suite's test defined like `func (s *notificationStorageTestSuite) Test_storage_Store()`. And I was within the body of Test_storage_Store() when I ran `:TestNearest`

**Comment by msvitok77** [comment]
https://github.com/vim-test/vim-test/assets/13028856/8e4ce274-adc5-4d96-9f86-1dd856c2d10a

I've prepared a simple video take a look

**Comment by FreedomKnight** [comment]
I know what you mean now.
Is that a bug from testify?
Or can you give me a right argument to build the command?

**Comment by msvitok77** [comment]
to command would be: `go test -timeout 30s -run ^TestNotificationStorageTestSuite$ -testify.m ^(Test_storage_Store)$ path/to/the/package -v`

**Comment by timsofteng** [comment]
@FreedomKnight Hi. Thank you for your contribution. How can I run testify suite test? Doesn't work for me. It cannot any nearest test.
